kong/conf_loader/init.lua

```diff
@@ -1856,6 +1856,55 @@ local function load(path, custom_conf, opts)
     end
   end
 
+  -- Wasm module support
+  if conf.wasm then
+    local wasm_filters = get_wasm_filters(conf.wasm_filters_path)
+    conf.wasm_modules_parsed = setmetatable(wasm_filters, _nop_tostring_mt)
+
+    local function add_wasm_directive(directive, value, prefix)
+      local directive_name = (prefix or "") .. directive
+      conf[directive_name] = conf[directive_name] or value
+    end
+
+    local wasm_main_prefix = "nginx_wasm_"
+
+    -- proxy_wasm_lua_resolver is intended to be 'on' by default, but we can't
+    -- set it as such in kong_defaults, because it can only be used if wasm is
+    -- _also_ enabled. We inject it here if the user has not opted to set it
+    -- themselves.
+    add_wasm_directive("nginx_http_proxy_wasm_lua_resolver", "on")
+
+    -- wasm vm properties are inherited from previously set directives
+    if conf.lua_ssl_trusted_certificate and #conf.lua_ssl_trusted_certificate >= 1 then
+      add_wasm_directive("tls_trusted_certificate", conf.lua_ssl_trusted_certificate[1], wasm_main_prefix)
+    end
+
+    if conf.lua_ssl_verify_depth and conf.lua_ssl_verify_depth > 0 then
+      add_wasm_directive("tls_verify_cert", "on", wasm_main_prefix)
+      add_wasm_directive("tls_verify_host", "on", wasm_main_prefix)
+      add_wasm_directive("tls_no_verify_warn", "on", wasm_main_prefix)
+    end
+
+    local wasm_inherited_injections = {
+      nginx_http_lua_socket_connect_timeout = "nginx_http_wasm_socket_connect_timeout",
+      nginx_proxy_lua_socket_connect_timeout = "nginx_proxy_wasm_socket_connect_timeout",
+      nginx_http_lua_socket_read_timeout = "nginx_http_wasm_socket_read_timeout",
+      nginx_proxy_lua_socket_read_timeout = "nginx_proxy_wasm_socket_read_timeout",
+      nginx_http_lua_socket_send_timeout = "nginx_http_wasm_socket_send_timeout",
+      nginx_proxy_lua_socket_send_timeout = "nginx_proxy_wasm_socket_send_timeout",
+      nginx_http_lua_socket_buffer_size = "nginx_http_wasm_socket_buffer_size",
+      nginx_proxy_lua_socket_buffer_size = "nginx_proxy_wasm_socket_buffer_size",
+      nginx_http_large_client_header_buffers = "nginx_http_wasm_socket_large_buffers",
+      nginx_proxy_large_client_header_buffers = "nginx_proxy_wasm_socket_large_buffers",
+    }
```

Comment:
Hmmm we don't have several of those directives: `proxy_wasm_socket_{read,connect,send}_timeout` and `proxy_wasm_socket_buffer_size`, `proxy_wasm_socket_large_buffers`. Our directives are `wasm_socket_*`, not `proxy_wasm_socket_*`. Am I missing something more obvious as to how this works?

Also, technically the `large_client_header_buffers` for `http{}` is for the clients, while `wasm_socket_large_buffers` is for upstream response (from HTTP dispatches). I think this one can be left untouched (we default it to `4 8192`, which can be tweaked by users if decided, but if users tweak their expected client's buffer sizes, it might not be appropriate for buffers on the upstream response side (HTTP dispatches). If you see a reason why we should map these 2 directives still, I'm all hears.

Reply:
> Hmmm we don't have several of those directives: proxy_wasm_socket_{read,connect,send}_timeout and proxy_wasm_socket_buffer_size, proxy_wasm_socket_large_buffers. Our directives are wasm_socket_*, not proxy_wasm_socket_*. Am I missing something more obvious as to how this works?

The `nginx_http_*` directive inject into the `http{}` block, while `nginx_proxy_*` injects into the `server {}` block. Since the Wasm directives support blocks, the code maps to both. The prefix gets stripped from the directive before the directive is injected.

As for `wasm_socket_large_buffers`, since that was included in the [original PR](https://github.com/Kong/kong/pull/11000) implementing some directives, I assumed it had been discussed - I will get this one removed!

Reply:
> The `nginx_http_*` directive inject into the `http{}` block, while `nginx_proxy_*` injects into the `server {}` block. Since the Wasm directives support blocks, the code maps to both. The prefix gets stripped from the directive before the directive is injected.

Ok!